annotator-core/src/main/java/edu/ucr/cs/riple/core/metadata/trackers/generatedcode/LombokTracker.java

```diff
@@ -0,0 +1,77 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.core.metadata.trackers.generatedcode;
+
+import edu.ucr.cs.riple.core.metadata.method.MethodDeclarationTree;
+import edu.ucr.cs.riple.core.metadata.trackers.MethodRegionTracker;
+import edu.ucr.cs.riple.core.metadata.trackers.Region;
+import edu.ucr.cs.riple.scanner.generatedcode.SourceType;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Tracker for extending potentially impacted regions for elements which will be use in generated
+ * code by <a href="https://projectlombok.org">Lombok</a>. Lombok automatically propagates
+ * {@code @Nullable} annotation on fields to getter methods, therefore, extends the set of
+ * potentially impacted regions to all callers of that method as well. This tracker, will include
+ * all callers of any method which uses an element in a lombok generated code. This will guarantee
+ * that {@link edu.ucr.cs.riple.core.evaluators.graphprocessor.ParallelConflictGraphProcessor} will
+ * catch any triggered errors by an annotation including all copied annotations by lombok as well.
+ */
+public class LombokTracker implements GeneratedRegionTracker {
+
+  /** Method region tracker to get potentially impacted regions of a method. */
+  private final MethodRegionTracker tracker;
+  /** Method declaration tree instance. */
+  private final MethodDeclarationTree methodDeclarationTree;
+
+  public LombokTracker(
+      MethodDeclarationTree methodDeclarationTree, MethodRegionTracker methodRegionTracker) {
+    this.methodDeclarationTree = methodDeclarationTree;
+    this.tracker = methodRegionTracker;
+  }
+
+  @Override
+  public Set<Region> extendWithGeneratedRegions(Set<Region> regions) {
+    return regions.stream()
+        // filter regions where are created by lombok
+        .filter(region -> region.sourceType.equals(SourceType.LOMBOK) && region.isOnMethod())
+        // find the corresponding method for the region.
+        .map(region -> methodDeclarationTree.findNode(region.member, region.clazz))
+        .filter(Objects::nonNull)
+        // get method location.
+        .map(methodNode -> methodNode.location)
+        // add potentially impacted regions for the collected methods.
+        .flatMap(
+            onMethod -> {
+              Optional<Set<Region>> ans = tracker.getRegions(onMethod);
+              return ans.isPresent() ? ans.get().stream() : Stream.of();
+            })
+        .collect(Collectors.toSet());
```

Comment:
This works on methods only and one hop, right? I.e. if our normal exploration adds `@Nullable` to field `Foo.bar` and `Foo` is `@Data`, our initial affected regions (_before_ this fix) will include `getBar()`. Then this chance makes sure we add all methods/fields/etc that would be affected by `getBar()` itself automatically becoming `@Nullable` due to Lombok. Correct?

This might not be enough to track all the changes to fields and methods of the inner class `Foo.Builder` or their impacted regions if the class is also annotated `@Builder`. Right?

Happy if the answer is no and we can do that other case as a _follow up_ PR, btw 🙂

Reply:
I looked at generated code by `@Builder` and I did not see any generated code getting reused again. But I think as discussed in the follow up PR for more tests, we can also resolve this issue if it comes up 🙂